### ISSUE TITLE
Trying AMI login in a background thread (MONO)

### ISSUE THREAD

**Issue by JobaDiniz** [AsterNET/AsterNET#39]
I'm running in MONO 4.2.2.30 - _linux_
If I try to login in a background thread, it doesn't work, do you know why?

```
var manager = new Manager(host, port, user, pass);
Task.Factory.StartNew(()=> manager.Login());
```

Asterisk debug: _Connect attempt from 'IP' unable to authenticate_
Net debug:

```
_AsterNET.Manager.TimeoutException: Timeout waiting for protocol identifier
  at AsterNET.Manager.ManagerConnection.login (Int32 timeout) <0xb3e08168 + 0x00618> in <filename unknown>:0
  at AsterNET.Manager.ManagerConnection.Login (Int32 timeout) <0xb3e07e88 + 0x0001b> in <filename unknown>:0
  at UCB.Service.UCBServer.LoginAsterisk () <0xb3e06e08 + 0x000f3> in <filename unknown>:0
```

In the same thread works:

```
var manager = new Manager(host, port, user, pass);
manager.Login()
```

Running on Windows it works just fine thou

**Comment by skrusty** [comment]
Does this work when not using mono?

**Comment by JobaDiniz** [comment]
Yes... it does...
One other thing, when using MONO, is that when manager tries to **reconnect** it does not send *user/password* and asterisk rejects the reconnection...

Unfortunately, project needs to run on MONO, so we're having a hard time :)

**Comment by Deantwo** [comment]
This is fixed in #140.
